packages/components/Callout/src/Callout.tsx

```diff
@@ -0,0 +1,30 @@
+import { requireNativeComponent } from 'react-native';
+import { IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { calloutName, ICalloutProps, ICalloutSlotProps, ICalloutType } from './Callout.types';
+import { useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { composable } from '@uifabricshared/foundation-composable';
+
+const RCTCallout = requireNativeComponent('RCTCallout');
+
+export const Callout = composable<ICalloutType>({
```

Comment:
Just because I'm curious, why do some of our components use "compose" and others use "composable"?

Reply:
if I understand correctly (and this is a good point that should be documented if it isn't, I didn't find it) the difference is 'compose' creates a HOC [using composable internally] where composable creates a functional component.  @jasonvmo any elaboration would be useful in distinguishing here :)

Reply:
I would say the main difference is that composable is sort of a base-class component that doesn't care about what theming/styling system is used, but provides a mechanism (useStyling) to inject one.  It just asks you to define the Slots and your hook logic in usePrepareProps. You can also provide a useStyling implementation (you shouldn't, see compose below) and the render function if you have >1 slot.  It's useful for FTZ because it has no actual visuals and is just a direct render of a purely behavioral native component.  

It allows for "old school" functional components to be described in their individual parts so that they can be recomposed without violating the rule of hooks and allows for eliding unnecessary vdom nodes.  Our philosophy is a little different from the web, here, FYI.

Compose's **primary** purpose is the injection of our idea of what the useStyling implementation should be into the composable pattern.  So to this end it's useful to look at useStying to see what it does:

```ts
// get the theme value from the context (or the default theme if it is not set)
  const theme = React.useContext(ThemeContext) || getTheme();
  // get the cache for this component from the theme
  const cache = getComponentCache(theme);

  // resolve the array of settings for these options
  lookupOverride = lookupOverride || props;
  const { settings, key } = getThemedSettings<IComponentSettings<IWithTokens<TSlotProps, TTokens>>, ITheme>(
    options.settings,
    theme,
    cache,
    baseKey,
    lookupOverride,
    _getSettingsFromTheme as any
  );

  // finish by processing the tokens and turning IComponentSettings into ISlotProps (this removes things like _overrides)
  return returnAsSlotProps(
    processTokens<TSlotProps, TTokens, ITheme>((props as unknown) as TTokens, theme, settings as any, options.resolvedTokens, key, cache)
  ) as IWithTokens<TSlotProps, TTokens>;
```

So it 
1. Is aware of our Theme object, because it calls looks at the ThemeContext.  This is a big thing.
2. It checks the theme's cache for resolved component slot props before doing any work, if none found....
3. Gets your settings object/function, processes it in response to the theme & applies overrides that you specify in your state 
4. resolves your tokens

So I would expect any component that wants to participate in our Theming & Styling system to have a composed version.

Whether Callout needs both a composed version & composable version, or just one or the other depends on the scenarios we expect it being used in.

Reply:
In short,
- Composable - Purely behavioral, but provides extensibility point for a styling system to be provided, allows for recomposition without overhead.
- Compose - Injects our theming system into Compose

Reply:
One of our "big ideas" was that the web team would take our implementation of compose and just write a useStyling implementation that makes more sense in ReactJS because it uses CSS, but they chose another route :)